airbyte-integrations/connectors/source-shopify/source_shopify/streams/base_streams.py

```diff
@@ -0,0 +1,677 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import logging
+from abc import ABC, abstractmethod
+from functools import cached_property
+from typing import Any, Callable, Dict, Iterable, Mapping, MutableMapping, Optional, Union
+from urllib.parse import parse_qsl, urlparse
+
+import pendulum as pdm
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+from requests.exceptions import RequestException
+from source_shopify.shopify_graphql.bulk.job import ShopifyBulkJob
+from source_shopify.shopify_graphql.bulk.query import BULK_PARENT_KEY, ShopifyBulkQuery, ShopifyBulkTemplates
+from source_shopify.transform import DataTypeEnforcer
+from source_shopify.utils import EagerlyCachedStreamState as stream_state_cache
+from source_shopify.utils import ShopifyNonRetryableErrors
+from source_shopify.utils import ShopifyRateLimiter as limiter
+
+
+class ShopifyStream(HttpStream, ABC):
+    # define default logger
+    logger = logging.getLogger("airbyte")
+
+    # Latest Stable Release
+    api_version = "2023-07"
+    # Page size
+    limit = 250
+
+    primary_key = "id"
+    order_field = "updated_at"
+    filter_field = "updated_at_min"
+
+    raise_on_http_errors = True
+    max_retries = 5
+
+    def __init__(self, config: Dict):
+        super().__init__(authenticator=config["authenticator"])
+        self._transformer = DataTypeEnforcer(self.get_json_schema())
+        self.config = config
+
+    @property
+    @abstractmethod
+    def data_field(self) -> str:
+        """The name of the field in the response which contains the data"""
+
+    @property
+    def url_base(self) -> str:
+        return f"https://{self.config['shop']}.myshopify.com/admin/api/{self.api_version}/"
+
+    @property
+    def default_filter_field_value(self) -> Union[int, str]:
+        # certain streams are using `since_id` field as `filter_field`, which requires to use `int` type,
+        # but many other use `str` values for this, we determine what to use based on `filter_field` value
+        # by default, we use the user defined `Start Date` as initial value, or 0 for `id`-dependent streams.
+        return 0 if self.filter_field == "since_id" else (self.config.get("start_date") or "")
+
+    def path(self, **kwargs) -> str:
+        return f"{self.data_field}.json"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        next_page = response.links.get("next", None)
+        if next_page:
+            return dict(parse_qsl(urlparse(next_page.get("url")).query))
+        else:
+            return None
+
+    def request_params(self, next_page_token: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
+        params = {"limit": self.limit}
+        if next_page_token:
+            params.update(**next_page_token)
+        else:
```

Comment:
It does not need to be set. The `updated_at_max` was introduced just to pass the CAT for `1.1.5` specifically, because of the large sandbox data. With this change, the CAT passes within an hour completely.